python/ml4ir/base/data/relevance_dataset.py

```diff
@@ -113,7 +116,7 @@ def create_dataset(self, parse_tfrecord=True):
                 parse_tfrecord=parse_tfrecord,
                 file_io=self.file_io,
                 logger=self.logger,
-            )
+            ).prefetch(tf.data.experimental.AUTOTUNE)
```

Comment:
I don't think this is worth adding a unit test (note that the code is exercised in integration tests).

Reply:
@mbrette I recommend making the change here instead -> https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/base/data/tfrecord_reader.py#L352

Reply:
Thoughts?

Reply:
I considered this, but then I was worried that it would not be exercised by tests, or not all tests.

Reply:
It should be. Everything that hits this code line will hit the same in tfrecord_reader as well. Luckily or unluckily.

Reply:
No, I meant that some tests use CSV dataset. It will not be exercised there

Reply:
It should be. The way ml4ir handles CSV is by converting it into TFRecords. Check here -> https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/base/data/csv_reader.py#L70

Reply:
Did you see any tests failing?

Reply:
the point where you suggest to put it is conditional to the user passing a `batch_size`. I would be OK with any choice. I would even suggest [this line](https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/base/data/tfrecord_reader.py#L361), but I don't think this is a blocker.

Reply:
@balikasg That was intentional. `prefetch` can only work if you have batched your dataset? What does prefetch even do otherwise? It will probably just use batch size = 1. That is not really what we are trying to do, right?

Reply:
I followed both your suggestions. prefetch is now performed as a last step in tfrecord_reader.